### ISSUE TITLE
Needs to get the default value of an input parameter in a resource function

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/ballerina-lang#30106]
**Description:**
When there is an input parameter with a default value in a resource function, there should be a way to get the default value of this parameter. This is a requirement for the Ballerina GraphQL package implementation.

Related: #27417, https://github.com/ballerina-platform/ballerina-standard-library/issues/1266

**Comment by hasithaa** [comment]
Do you need this during the runtime or compile time? Default values will be captured as closures. So you can't get a value unless it is a constant value during the compile time. At runtime, you have to manually evaluate the closure to get the value. But that is dangerous because closures can mutate the state. 

Can you summarize the use case here?

**Comment by ThisaruGuruge** [comment]
Currently, we create the GraphQL schema for a given service at the runtime, since we don't have a direct way to create the schema at the compile time and propagate that to the runtime.

The use case of this specific scenario is that we need the default values of the input parameters to support GraphQL introspection. When a resource function has an input parameter, it maps to an InputValue type in GraphQL. If there is a default value for that input value, GraphQL clients should be able to query the GraphQL endpoint and view the default value of a given input value. Currently, we do not have a way to get this default value. (https://github.com/ballerina-platform/ballerina-standard-library/issues/1377)

**Comment by ThisaruGuruge** [comment]
Any update on this?

**Comment by rdhananjaya** [comment]
@chiranSachintha Can we do this after function default value change? I mean technically jBallerina backend team only need to provide an API to access the closures right?

**Comment by ThisaruGuruge** [comment]
Hi, any update on this?

**Comment by rdhananjaya** [comment]
@warunalakshitha can you look into this?

**Comment by ThisaruGuruge** [comment]
Any update on this? We cannot fix https://github.com/ballerina-platform/ballerina-standard-library/issues/1377 without this.

**Comment by gimantha** [comment]
> Any update on this? We cannot fix [ballerina-platform/ballerina-standard-library#1377](https://github.com/ballerina-platform/ballerina-standard-library/issues/1377) without this.

We are still working on passing default values to runtime( Can be tracked via https://github.com/ballerina-platform/ballerina-lang/issues/36020) We will update on the progress

**Comment by warunalakshitha** [comment]
Closing the issue since https://github.com/ballerina-platform/ballerina-lang/issues/36020 is done.
